### PR TITLE
Check if context is cancelled before next iteration

### PR DESCRIPTION
**Pull request by dharmicksai**
When the health check fails, the test keeps on retrying.
If the user tries to stop the test prematurely by sending a SIGINT signal,
the pod.K8sClient.ExecInPod function returns an error, as the context is canceled.
due to this, the test would get stuck in an infinite loop.

Now the context is checked for cancellation before continuing to the next loop

Fixes: #1314

Signed-off-by: K Dharmick Sai <dharmicksaik@gmail.com>

### CODE REVIEW

File: connectivity/tests/health.go [hunk 1]
```diff
@@ -45,8 +45,13 @@ func runHealthProbe(ctx context.Context, t *check.ConnectivityTest, pod *check.P
 
 		stdout, err := pod.K8sClient.ExecInPod(ctx, pod.Pod.Namespace, pod.Pod.Name, "", cmd)
 		if err != nil {
-			t.Warnf("cilium-health probe failed: %q, stdout: %q, retrying...", err, stdout)
-			continue
+			select {
+			case <-done:
```
Comment (bimmlerd) [inline connectivity/tests/health.go:49]: this fixes the immediate bug regarding ctx cancellation through Ctrl+C, but as mentioned in https://github.com/cilium/cilium-cli/issues/1314#issuecomment-1376929199 we should also check the retryTimer here.

**Review by bimmlerd — CHANGES_REQUESTED** [review]
Thanks for your contribution!
Approach looks good, just need to also check the retryTimer.

File: connectivity/tests/health.go (hunk 1 above)
Comment (dharmicksai) [reply connectivity/tests/health.go:49]: Oh, I will update the pr

**Comment by dharmicksai** [comment]
Multicluster / installation-and-connectivity (pull_request_target)  check is failing with the following error messages:

`❌ command "curl -w %{local_ip}:%{local_port} -> %{remote_ip}:%{remote_port} = %{response_code} --silent --fail --show-error --output /dev/null --connect-timeout 5 --max-time 20 http://echo-other-node:8080" failed: command terminated with exit code 28`
`❌ Aborting flow matching: context deadline exceeded` 
`❌ Flow validation failed for peer cilium-test/client-755fb678bd-wzxzh: 1 failures (first: 0, last: 8, matched: 2)`

The errors are being logged in the [ValidateFlows function](https://github.com/cilium/cilium-cli/blob/v0.12.12/connectivity/check/action.go#L825)
 How can these errors be resolved?

File: connectivity/tests/health.go [hunk 2]
```diff
@@ -45,21 +45,14 @@ func runHealthProbe(ctx context.Context, t *check.ConnectivityTest, pod *check.P
 
 		stdout, err := pod.K8sClient.ExecInPod(ctx, pod.Pod.Namespace, pod.Pod.Name, "", cmd)
 		if err != nil {
-			select {
-			case <-done:
+			t.Warnf("cilium-health probe failed: %q, stdout: %q, retrying...", err, stdout)
+		}else{
```
Comment (bimmlerd) [inline connectivity/tests/health.go:49]: nit: looks like you didn't run gofmt, please do.

**Review by bimmlerd — CHANGES_REQUESTED** [review]
Changes LGTM (except for gofmt nit). To get it merged, please squash the two commits and add a brief commit message describing the bug this fixes.

Regarding the CI, seems like a flake. Squashing the commits will retrigger it anyway, let's see whether it works ok then.

File: connectivity/tests/health.go (hunk 2 above)
Comment (dharmicksai) [reply connectivity/tests/health.go]: I ran gofmt and squashed the commits

**Comment by bimmlerd** [comment]
Change now lgtm. CI bombed pretty hard, but I'll retrigger those which I can see are unrelated:
- rerunning EKS (tunnel): https://github.com/cilium/cilium-cli/actions/runs/4044003310/jobs/6953620920#step:12:34 (something about helm.cilium.io)
- rerunning EKS (ENI): https://github.com/cilium/cilium-cli/actions/runs/4044003304/jobs/6953620932#step:12:33 (something about helm.cilium.io)\
- the others failed with "no agent available" which also sounds like an infra issue.

will approve once CI is happier.

**Review by bimmlerd — APPROVED** [review]
Had to rebase on latest master to get CI to pass.
(Seems like you rebased on `master` instead of `origin/master` without pulling `origin/master` into your local `master` first, thus resetting the commits back a couple of weeks, apparently breaking CI)
Thanks for sticking with this, looking forward to more contributions :)

**Comment by dharmicksai** [comment]
Thank you @bimmlerd, for guiding me through my first issue.

**Pull request merged**